packages/libs/sdk-mixins/src/mixins/staticResourcesMixin/fetchWithFallbacks.ts

```diff
@@ -0,0 +1,40 @@
+import { Logger } from '../loggerMixin';
+
+type FetchParams = Parameters<typeof fetch>;
+const notLastMsgSuffix = 'Trying the next fallback URL...';
+
+export const fetchWithFallbacks = async (
+  fallbacks: FetchParams['0'] | FetchParams['0'][],
+  init: FetchParams['1'],
+  {
+    logger,
+    onSuccess,
+  }: { logger?: Logger; onSuccess?: (urlIndex: number) => void },
+): ReturnType<typeof fetch> => {
+  const fallbacksArr = Array.isArray(fallbacks) ? fallbacks : [fallbacks];
+
+  for (let index = 0; index < fallbacksArr.length; index++) {
+    const url = fallbacksArr[index];
+    const isLast = index === fallbacksArr.length - 1;
+
+    try {
+      const res = await fetch(url, init);
+      if (res.ok) {
+        onSuccess?.(index);
```

Comment:
might be just a personal opinion, but I think that the fact that the `onSuccess` can be replaced with a complex response (`{ res: ReturnType<typeof fetch>;  urlIndex: number }`)

IMHO its simpler to return such type than having two ways to interact with this function 
(if we would use fetch and return value in different places that is something else, but its not the case here)

this is just my opinion ofc

Reply:
Decided to keep the current impl